_targets.R

```diff
@@ -0,0 +1,39 @@
+library(targets)
+source("1_fetch/src/download.R")
+source("2_process/src/load.R")
+source("3_visualize/src/plot.R")
+
+tar_option_set(packages = c("tidyverse", "sbtools", "whisker", "dplyr"))
+
+list(
+  # Get the data from ScienceBase
+  tar_target(
+    model_RMSEs_csv,
+    download_data(out_filepath = "1_fetch/out/model_RMSEs.csv"),
+    format = "file"
+  ), 
+  # Prepare the data for plotting
+  tar_target(
+    eval_data,
+    load_RMSE(in_filepath = model_RMSEs_csv),
+    packages = c("readr", "stringr")
+  ),
+  # Create a plot
+  tar_target(
+    test_RMSE_png,
+    plot_test_RMSE(eval_data, out_filepath = "3_visualize/out/test_RMSE.png"), 
+    format = "file"
+  ),
+  # Save the processed data
+  tar_target(
+    model_summary_results_csv,
+    save_RMSE_csv(eval_data, out_filepath = "2_process/out/model_summary_results.csv"), 
+    format = "file"
+  ),
```

Comment:
This approach (of calling a function that writes the csv and returns the filepath) works great, but just wanted to alert  you to another option for targets that only require a simple action like writing an existing dataframe to csv. Rather than calling a function to build the target, you can actually write the code to build the target into the target definition. It can feel a bit clunky for file targets, since you need to return the filepath, but it does work, and can be handy for small steps, e.g. filtering a dataframe object target:
```r
# Save the processed data
  tar_target(
    model_summary_results_csv,
    {
      out_file = "2_process/out/model_summary_results.csv"
      readr::write_csv(eval_data, 
                file = out_file)
      return(out_file)
    }, 
    format = "file"
  ),
```
If it's just a single line/pipe of code, you don't need those curly braces `{ }`:
```r
  tar_target(
    raw_data_filtered,
    read_csv(model_rmses_csv, col_types = cols()) %>%
      filter(exper_n == 1)
  ),
```